x-pack/plugins/apm/server/routes/settings/agent_configuration/route.ts

```diff
@@ -38,7 +39,26 @@ const agentConfigurationRoute = createApmServerRoute({
     >;
   }> => {
     const setup = await setupRequest(resources);
-    const configurations = await listConfigurations({ setup });
+
+    const [agentConfigAppliedByEtag, configList] = await Promise.all([
```

Comment:
This could also be stored to`apmAgentConfigurationIndex`, otherwise `applied_by_agent` field will refer only to configs applied to agents using standalone APM server. Since this index is not updated by then APM server does it make sense to do it from the agent config list page or through a background task? cc @dgieselaar

Reply:
Updated in https://github.com/elastic/kibana/pull/131820/commits/7329fe0e820299137a9a42ca76ec9bbfe24c28f5 - this could also be a background task